css/css-display/display-contents-accessibility.html

```diff
@@ -0,0 +1,321 @@
+
+    <button aria-label="label" style="display: contents;" data-expectedlabel="label" data-testname="button with display:contents is labelled via aria-label" class="ex-label"></button>
+    <div role="group" aria-label="label" style="display: contents;" data-expectedlabel="label" data-testname="div with role group with display:contents is labelled via aria-label" class="ex-label"></div>
+
+    <!-- <img> renders "display:contents" the same as "display:none", see: https://www.w3.org/TR/css-display-3/#unbox-html-->
+    <label for="label-for-button">label</label>
+    <button id="label-for-button" style="display: contents;" data-expectedlabel="label" data-testname="button with display:contents is labelled via label for/id association" class="ex-label"></button>
+    <!-- SVG and display: contents behavior: https://github.com/w3c/svgwg/issues/305-->
+    <svg viewbox="0 0 100 50">
+        <g style="display:contents;" fill="silver" stroke="blue" data-expectedlabel="group label" data-testname="g element with display:contents, as child of svg, is labelled via title element" class="ex-label">
+          <title>group label</title>
+          <circle cx="20" cy="20" r="10" />
+          <circle cx="20" cy="20" r="10" />
+        </g>
+      </svg>
+
+    <a href="#" style="display: contents;" data-expectedlabel="label" data-testname="link with display:contents is labelled via name from contents" class="ex-label">label</a>
+    <button style="display: contents;" data-expectedlabel="label" data-testname="button with display:contents is labelled via name from contents" class="ex-label">label</button>
+    <h3 style="display: contents;" data-expectedlabel="label" data-testname="h3 with display:contents is labelled via name from contents" class="ex-label">label</h3>
+
+    <h2>Accessibility roles</h2>
+    <!-- Buttons -->
+    <button style="display: contents;" data-expectedrole="button" data-testname="button with display:contents has button role" class="ex">x</button>
+    <button role="button" style="display: contents;" data-expectedrole="button" data-testname="button with explicit button role and display:contents has button role" class="ex">x</button>
+    <div role="button" style="display: contents;" data-expectedrole="button" data-testname="div with button role and display:contents has button role" class="ex">x</div>
+    <div role="button" tabindex="0" style="display: contents;" data-expectedrole="button" data-testname="div with button role, tabindex=0 and display:contents has button role" class="ex">x</div>
+
+    <!-- Display: Flex -->
+    <table style="display: flex; flex-flow: column nowrap;">
+        <thead>
+          <tr style="display: contents;" data-expectedrole="row" data-testname="tr with display:contents, within table with display:flex, has row role" class="ex">
+            <th data-expectedrole="rowheader" data-testname="th as child of tr with display:contents, within table with display:flex, has rowheader role" class="ex">x</th>
```

Comment:
Re: "arguable" I think it's reasonable that the top left `th` is still considered a columnheader (despite the explicit role, it's really both). You could also "fix" this test by moving that particular check to a `th` in a later row. I'm guessing that would pass in both Gecko and Chromium.

Reply:
I must've imagined the explicit role, because (`role="rowheader"`) isn't there. So another fix would be to change the expectation to `data-expectedrole="columnheader"`) w/ or perhaps w/o an explicit role attr.

Reply:
@cookiecrook @zcorpan Thank you, I've changed this test to expect `role="columnheader"`.

> (@cookiecrook) All browsers fail this test. WebKit's error message (!EQ("rowheader", "")) is probably a bug covered by another test but the Gecko and Chromium failures (!EQ("rowheader", "columnheader")) seem like a separate issue that is both arguable and unrelated to display:contents.

Here is the existing `<th>` role test in [wpt/html-aam/table-roles.html](https://github.com/web-platform-tests/wpt/blob/master/html-aam/table-roles.html) and its results: https://wpt.fyi/results/html-aam/table-roles.html?label=experimental&label=master&aligned. It also looks like `<th>` role calculation also depends on whether it is in a `<thead>` or `<tbody>` context.